corehq/apps/commtrack/views.py

```diff
@@ -316,23 +316,30 @@ def get_server_date_by_form_id(self, form_id):
         else:
             return ServerTime(server_date).ui_string()
 
+    def _get_selected_case_id(self):
+        location_id = self.request.GET.get('location_id')
+        if location_id:
+            try:
+                return (SQLLocation.objects
+                        .get(domain=self.domain, location_id=location_id)
+                        .supply_point_id)
+            except SQLLocation.DoesNotExist:
+                messages.error(self.request, 'Your location id did not match a location')
+
     @property
     def page_context(self, **kwargs):
-        stock_state_limit = 100
-        stock_transaction_limit = 10000
+        stock_state_limit = self.request.GET.get('stock_state_limit', 100)
+        stock_transaction_limit = self.request.GET.get('stock_transaction_limit', 1000)
         stock_state_limit_exceeded = False
         stock_transaction_limit_exceeded = False
 
         query = StockTransaction.objects.filter(report__domain=self.domain)
-        if self.location_id:
-            try:
-                case_id = (SQLLocation.objects
-                           .get(domain=self.domain, location_id=self.location_id)
-                           .supply_point_id)
-            except SQLLocation.DoesNotExist:
-                messages.error(self.request, 'Your location id did not match a location')
-            else:
-                query = query.filter(case_id=case_id)
+        selected_case_id = self._get_selected_case_id()
+        if selected_case_id:
+            query = query.filter(case_id=selected_case_id)
+        selected_product_id = self.request.GET.get('product_id')
+        if selected_product_id:
+            query = query.filter(product_id=selected_product_id)
```

Comment:
is this filtering new?

Reply:
Yup, adding it in this PR.  Figured it'd help drilling down to specific issues.